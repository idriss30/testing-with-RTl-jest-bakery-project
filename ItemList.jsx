import React from "react";

export const ItemList = ({ items }) => {
  return (
    <ul>
      {Object.entries(items).map(([itemName, quantity]) => {
        return (
          <li key={itemName}>
            {itemName} - Quantity : {quantity}
          </li>
        );
      })}
    </ul>
  );
};
