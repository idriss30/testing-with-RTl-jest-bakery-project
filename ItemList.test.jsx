import React from "react";
import { render } from "@testing-library/react";
import { ItemList } from "./ItemList.jsx";

describe("testing ItemsListComponents", () => {
  test("render list of items with ItemList", () => {
    const items = [
      { productName: "cheesecake", productQty: 3 },
      { productName: "danish", productQty: 5 },
      { productName: "croissant", productQty: 20 },
    ];
    const { getByText } = render(<ItemList items={items} />);
    const nodeList = document.querySelector("ul");
    expect(nodeList.childElementCount).toBe(3);
    expect(getByText("cheesecake - Quantity : 3")).toBeInTheDocument();
    expect(getByText("danish - Quantity : 5")).toBeInTheDocument();
    expect(getByText("croissant - Quantity : 20")).toBeInTheDocument();
  });
});
