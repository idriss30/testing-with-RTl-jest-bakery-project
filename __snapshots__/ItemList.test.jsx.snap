// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing style for items that are less or equal to 5 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0;
  }

  50% {
    opacity: .5;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  font-weight: bold;
  color: crimson;
  -webkit-animation: animation-0 3s linear infinite;
  animation: animation-0 3s linear infinite;
}

<li
  class="emotion-0"
>
  cheesecake - Quantity : 4
</li>
`;

exports[`testing style for items that are less or equal to 5 2`] = `
@keyframes animation-0 {
  0% {
    opacity: 0;
  }

  50% {
    opacity: .5;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  font-weight: bold;
  color: crimson;
  -webkit-animation: animation-0 3s linear infinite;
  animation: animation-0 3s linear infinite;
}

<li
  class="emotion-0"
>
  croissant - Quantity : 2
</li>
`;
